algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -38,33 +38,73 @@ package com.twitter.algebird
  */
 
 object QTree {
-  def apply[A: Monoid](kv: (Double, A), level: Int = -16): QTree[A] = {
+  /**
+   * level gives a bin size of 2^level. By default the bin size is 1/65536 (level = -16)
+   */
+  def apply[A](kv: (Double, A), level: Int = -16): QTree[A] =
     QTree(math.floor(kv._1 / math.pow(2.0, level)).toLong,
       level,
       1,
       kv._2,
       None,
       None)
-  }
 
-  def apply[A: Monoid](kv: (Long, A)): QTree[A] = {
+  def apply[A](kv: (Long, A)): QTree[A] =
     QTree(kv._1,
       0,
       1,
       kv._2,
       None,
       None)
-  }
 
   /**
-   * The common case of wanting a count and sum for the same value
+   * The common case of wanting an offset and sum for the same value
+   * This is useful if you want to query the mean inside a range later.
+   * If you truly just care about the counts/histogram, see the value method.
    */
   def apply(k: Long): QTree[Long] = apply(k -> k)
+  /**
+   * uses 1/65636 as the bin size, if you want to control that see other apply
+   * or value methods.
+   *
+   * This is useful if you want to query the mean inside a range later.
+   * If you truly just care about the counts/histogram, see the value method.
+   */
   def apply(k: Double): QTree[Double] = apply(k -> k)
+
+  /**
+   * If you are sure you only care about the approximate histogram
+   * features of QTree, you can save some space by using QTree[Unit]
+   */
+  def value(v: Long): QTree[Unit] = apply(v -> ())
+  /**
+   * If you are sure you only care about the approximate histogram
+   * features of QTree, you can save some space by using QTree[Unit]
+   * level gives a bin size of 2^level. By default this is 1/65536 (level = -16)
+   */
+  def value(v: Double, level: Int = -16): QTree[Unit] = apply(v -> (), level)
 }
 
-class QTreeSemigroup[A: Monoid](k: Int) extends Semigroup[QTree[A]] {
+class QTreeSemigroup[A](k: Int)(implicit val underlyingMonoid: Monoid[A]) extends Semigroup[QTree[A]] {
+  /** Override this if you want to change how frequently sumOption calls compress */
+  def compressBatchSize: Int = 25
   def plus(left: QTree[A], right: QTree[A]) = left.merge(right).compress(k)
+  override def sumOption(items: TraversableOnce[QTree[A]]): Option[QTree[A]] = if (items.isEmpty) None
+  else {
+    // only call compressBatchSize once
+    val batchSize = compressBatchSize
+    var count = 1 // start at 1, so we only compress after batchSize items
```

Comment:
That's not the case though.  If batchSize = 2, that condition will hold on the first pass through the loop, not the second.

It's a little weird to think about.  Here, the first compression will be on an uncompressed combination of 25 items.  Other compressions will be on 26 (including the previous compression), until the last one.  Initializing count to 1 if it avoids doing a compression on the _first_ iteration makes sense to me.  But 25 vs. 26 for the first compression only seems inconsequential.